core/src/main/java/hudson/cli/DeleteItemCommand.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2015 Red Hat, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package hudson.cli;
+
+import hudson.AbortException;
+import hudson.Extension;
+import hudson.model.AbstractItem;
+import hudson.model.Node;
+import jenkins.model.Jenkins;
+import org.kohsuke.args4j.Argument;
+
+import java.util.List;
+import java.util.HashSet;
+
+/**
+ * CLI command, which helps delete job(s) or node(s).
+ * @author noahwc
+ * @since TODO
+ */
+
+public abstract class DeleteItemCommand extends CLICommand {
```

Comment:
This needs a different name, perhaps `DeleteCommand`? `Item` already exists and is a supertype of `Job`; in fact, `delete-job` can already delete items that aren't jobs (so is a misnomer now).